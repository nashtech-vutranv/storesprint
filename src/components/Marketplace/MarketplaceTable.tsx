import React, {useState, useEffect, useContext} from 'react'
import {useLocation, useNavigate, useSearchParams} from 'react-router-dom'
import {useTranslation} from 'react-i18next'
import {Column} from 'primereact/column'
import {Button as ButtonPrime} from 'primereact/button'
import {DataTable, DataTableSelectionChangeParams} from 'primereact/datatable'
import {Button, Row, Col} from 'react-bootstrap'
import {GlobalContext} from '../../store/GlobalContext'
import {usePagination, useCommonAccesibility, useHandleError, usePreviousPage} from '../../hooks'
import MarketplaceService from '../../services/MarketplaceService'
import {capitalizeFirstLetter} from '../../helpers/characters'
import {TemplatePaginator} from '../Paginator'
import {ROUTE_MARKETPLACE, ROUTE_PARAMS, PERMISSIONS} from '../../constants'
import FieldTextDataTable from '../FieldTextDataTable/FieldTextDataTable'
import {MarketplaceActionType, RowTableActionType} from '../../store/actions'
import {defaultMarketplacesTablePaginationPerPage} from '../../constants/pagination'
import {IMarketplace} from '../../interface/marketplace'
import {AddButton} from '../Common'
import {getURLParamsObj} from '../../helpers'
import {IUrlParams} from '../../interface'
import Searching from '../Searching'

export default function MarketplaceTable() {
  const {t} = useTranslation()
  const navigate = useNavigate()
  const location = useLocation()
  const [selectedURLParamsObj, setSelectedURLParamsObj] = useState<IUrlParams>(
    getURLParamsObj()
  )

  const [, setSearchParams] = useSearchParams()

  const {
    state: {axiosClient},
    dispatch: {marketplace: marketplaceDispatch, rowTable: rowTableDispatch},
  } = useContext(GlobalContext)

  const {handleErrorResponse} = useHandleError()

  const {
    dataApi: {keyword, pagination, defaultPagination},
    dataTable: {
      totalRecords,
      setTotalRecords,
      onPage,
      onSort,
      onSearch,
      onKeywordChange,
      setPagination,
      setKeyword,
      onBlurInputSearch,
    },
  } = usePagination(defaultMarketplacesTablePaginationPerPage, true)

  const [marketplaces, setMarketplaces] = useState<IMarketplace[]>([])

  useEffect(() => {
    if (pagination) {
      new MarketplaceService(axiosClient)
        .fetchAllMarketplaces(pagination, {
          search: keyword,
        })
        .then((response: any) => {
          const mappingData = response.data
          const marketplacesData = mappingData.content.map((item: any) => ({
            ...item,
            status: capitalizeFirstLetter(item.status),
          }))
          setMarketplaces(marketplacesData)
          setTotalRecords(mappingData.totalElements)
        })
        .catch((err: any) => {
          handleErrorResponse(err)
        })
    }
  }, [pagination])

  useEffect(() => {
    setSearchParams(selectedURLParamsObj as any)
  }, [selectedURLParamsObj])

  useEffect(() => {
    pagination &&
      setSelectedURLParamsObj({
        ...selectedURLParamsObj,
        search: pagination.keyword,
        page: pagination.page,
        sortField: pagination.sortField,
        sortOrder: pagination.sortOrder,
        first: pagination.first,
      })
  }, [pagination])

  usePreviousPage('apps-marketplaces', {})

  useCommonAccesibility()

  const refreshInitialState = () => {
    setKeyword('')
    pagination &&
      setPagination({
        ...defaultPagination,
        sortField: pagination?.sortField,
        sortOrder: pagination?.sortOrder,
      })
    rowTableDispatch({
      type: RowTableActionType.GET_CURRENT_TABLE_ROW_QUANTITIES,
      payload: 10,
    })
  }

  const onSelectionChange = (e: DataTableSelectionChangeParams) => {
    marketplaceDispatch({
      type: MarketplaceActionType.GET_MARKETPLACE_INFORMATION,
      payload: e.value,
    })
    navigate(
      ROUTE_MARKETPLACE.EDIT.replace(ROUTE_PARAMS.MARKETPLACE_ID, e.value.id),
      {
        state: {
          viewlistLocation: location,
        },
      }
    )
  }

  const onPressKeyDown = (event: any) => {
    if (event.key === 'Enter') {
      onSearch()
    }
  }

  const renderHeader = () => {
    return (
      <Row>
        <div className='d-flex justify-content-start align-items-center'>
          <Col xs={8} sm={8} md={8} lg={8} xl={8} xxl={8} className='btn-mr-1'>
            <Searching
              value={keyword}
              setKeyword={setKeyword}
              onKeywordChange={onKeywordChange}
              placeholder={t('common_input_search_placeHolder')}
              onBlurInputSearch={onBlurInputSearch}
              onPressKeyDown={onPressKeyDown}
              pagination={pagination}
              setPagination={setPagination}
            />
          </Col>
          <Button onClick={onSearch} className='btn-h-95'>
            {t('common_button_search_label')}
          </Button>
        </div>
      </Row>
    )
  }

  const templateLamda = TemplatePaginator('sample_lamda')

  const paginatorLeft = (
    <ButtonPrime
      aria-label='refresh-button'
      type='button'
      icon='pi pi-refresh'
      className='p-button-text refresh-button'
      onClick={refreshInitialState}
      tooltip={t('common_button_refresh_table')}
    />
  )

  const paginatorRight = (
    <AddButton
      onClick={() => navigate(ROUTE_MARKETPLACE.ADD)}
      permissions={[PERMISSIONS.add_marketplace]}
      label={t('marketplace_paginator_add_marketplace')}
    />
  )

  return (
    <>
      <DataTable
        sortField={pagination?.sortField}
        sortOrder={pagination?.sortOrder}
        totalRecords={totalRecords}
        dataKey='id'
        value={marketplaces}
        paginator
        paginatorTemplate={templateLamda}
        first={pagination?.first}
        rows={pagination?.rows}
        onSelectionChange={(e) => onSelectionChange(e)}
        responsiveLayout='scroll'
        selectionMode='single'
        header={renderHeader}
        className='data-table-mh'
        paginatorClassName='table-paginator'
        emptyMessage={t('marketplace_table_empty_message')}
        paginatorLeft={paginatorLeft}
        paginatorRight={paginatorRight}
        lazy
        onSort={onSort}
        onPage={onPage}
      >
        <Column
          field='name'
          header={t('marketplace_column_header_name')}
          sortable
          body={(agg: IMarketplace) => (
            <FieldTextDataTable value={agg.name} placement='bottom' />
          )}
        ></Column>
        <Column
          field='currencyId'
          header={t('marketplace_column_header_currency')}
          sortable
          sortField='currency.name'
          body={(agg: IMarketplace) => (
            <FieldTextDataTable value={agg.currency?.name} placement='bottom' />
          )}
        ></Column>
        <Column
          field='status'
          header={t('marketplace_column_header_status')}
          sortable
        ></Column>
      </DataTable>
    </>
  )
}
