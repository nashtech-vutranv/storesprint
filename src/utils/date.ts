import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
