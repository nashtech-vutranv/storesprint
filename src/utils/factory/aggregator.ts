import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
