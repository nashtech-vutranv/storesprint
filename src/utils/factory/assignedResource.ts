import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
