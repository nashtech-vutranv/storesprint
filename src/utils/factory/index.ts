export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
