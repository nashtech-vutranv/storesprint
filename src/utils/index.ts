export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
