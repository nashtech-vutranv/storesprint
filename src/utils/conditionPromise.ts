import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}