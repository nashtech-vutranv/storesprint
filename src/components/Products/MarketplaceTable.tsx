import {DataTable} from 'primereact/datatable'
import {useContext, useEffect, useState} from 'react'
import {useTranslation} from 'react-i18next'
import _ from 'lodash'
import {Button as ButtonPrime} from 'primereact/button'
import {Column} from 'primereact/column'
import {defaulMarketplaceTablePaginationPerPage} from '../../constants/pagination'
import {capitalizeFirstLetter} from '../../helpers'
import {TemplatePaginator} from '../Paginator'
import {RowTableActionType} from '../../store/actions'
import ProductServices from '../../services/ProductService'
import {GlobalContext} from '../../store/GlobalContext'
import FieldTextDataTable from '../FieldTextDataTable/FieldTextDataTable'
import {usePagination, useHandleError} from '../../hooks'
import {formatDate} from '../../utils'

export interface IMarketplaceProductTableProps {
  productId?: string
}

export default function MarketplaceTable({productId}: IMarketplaceProductTableProps) {
  const {t} = useTranslation()
  const [marketplaceProducts, setMarketplaceProducts] = useState<any[]>([])

  const {
    state: {axiosClient},
    dispatch: {rowTable: rowTableDispatch},
  } = useContext(GlobalContext)

  const {handleErrorResponse} = useHandleError()

  const {
    dataApi: {pagination, defaultPagination},
    dataTable: {totalRecords, setTotalRecords, onPage, onSort, setPagination},
  } = usePagination(defaulMarketplaceTablePaginationPerPage)

  const getMarketplaceProductById = () => {
    if (productId) {
      new ProductServices(axiosClient)
        .getMarketplacesByProductId(productId, _.omit(pagination, ['keyword']))
        .then((response) => {
          setMarketplaceProducts(
            response.data.content.map((item: any) => ({
              ...item,
              createdAt: formatDate(item.createdAt)
            }))
          )
          setTotalRecords(response.data.totalElements)
        })
        .catch((err: any) => {
          handleErrorResponse(err)
        })
    }
  }

  const refreshInitialState = () => {
    pagination &&
      setPagination({
        ...defaultPagination,
        sortField: pagination?.sortField,
        sortOrder: pagination?.sortOrder,
      })
    rowTableDispatch({
      type: RowTableActionType.GET_CURRENT_TABLE_ROW_QUANTITIES,
      payload: 10,
    })
  }

  const templateLamda = TemplatePaginator('sample_lamda')

  const paginatorLeft = (
    <ButtonPrime
      aria-label='refresh-button'
      type='button'
      icon='pi pi-refresh'
      className='p-button-text refresh-button'
      onClick={refreshInitialState}
      tooltip={t('common_button_refresh_table')}
    />
  )

  useEffect(() => {
    pagination && productId && getMarketplaceProductById()
  }, [productId, pagination])

  useEffect(() => {
    document.querySelectorAll('.p-dropdown-trigger').forEach((el) => {
      el.setAttribute('aria-label', 'dropdown-trigger')
    })
  }, [document.querySelectorAll('.p-dropdown-trigger')])

  return (
    <>
      <DataTable
        sortField={pagination?.sortField}
        sortOrder={pagination?.sortOrder}
        totalRecords={totalRecords}
        dataKey='id'
        value={marketplaceProducts}
        first={pagination?.first}
        rows={pagination?.rows}
        paginator
        paginatorTemplate={templateLamda}
        paginatorClassName='table-paginator'
        responsiveLayout='scroll'
        selectionMode='single'
        className='data-table-mh mt-4'
        emptyMessage={t('product_detail_table_no_data')}
        paginatorLeft={paginatorLeft}
        lazy
        onSort={onSort}
        onPage={onPage}
      >
        <Column
          header={t('product_detail_marketplace_table_marketplace')}
          sortable
          sortField='localeMarketplace.marketplace.name'
          body={(data: any) => (
            <FieldTextDataTable
              value={data.marketplace.name}
              placement='bottom'
            />
          )}
        ></Column>
        <Column
          header={t('product_detail_marketplace_table_listed_price')}
          sortable
          sortField='sellingPrice'
          body={(data: any) => (
            <FieldTextDataTable value={data.sellingPrice} placement='bottom' />
          )}
        ></Column>
        <Column
          header={t('product_detail_marketplace_table_currencry')}
          sortable
          sortField='localeMarketplace.marketplace.currency.id'
          body={(data: any) => (
            <FieldTextDataTable value={data.currency.id} placement='bottom' />
          )}
        ></Column>
        <Column
          header={t('product_detail_marketplace_table_quantity')}
          sortable
          sortField='listingQuantity'
          body={(data: any) => (
            <FieldTextDataTable
              value={data.listingQuantity}
              placement='bottom'
            />
          )}
        ></Column>
        <Column
          header={t('product_detail_marketplace_table_listed_date')}
          sortable
          sortField='createdAt'
          body={(data: any) => (
            <FieldTextDataTable value={data.createdAt} placement='bottom' />
          )}
        ></Column>
        <Column
          header={t('product_detail_marketplace_table_status')}
          sortable
          sortField='listingState.status'
          body={(data: any) => (
            <FieldTextDataTable
              value={capitalizeFirstLetter(data.listingState.status)}
              placement='bottom'
            />
          )}
        ></Column>
        <Column
          header={t('product_detail_marketplace_table_failed_reason')}
          sortable
          sortField='listingState.errorMessage'
          body={(data: any) => (
            <FieldTextDataTable
              value={data.listingState.errorMessage}
              placement='bottom'
            />
          )}
        ></Column>
      </DataTable>
    </>
  )
}
