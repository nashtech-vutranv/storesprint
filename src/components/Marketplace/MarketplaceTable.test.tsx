import {render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import {mockMarketplaces} from '../../utils/factory'
import MarketplaceTable from './MarketplaceTable'

jest.mock('../../services/MarketplaceService', () => {
  const instance = jest.fn().mockReturnValue({
    fetchAllMarketplaces: jest.fn().mockImplementation(() => {
      return Promise.resolve(mockMarketplaces())
    }),
  })

  return instance
})

describe('Marketplace Table', () => {
    it('should match snapshot', async () => {
      const tree = render(
        <MemoryRouter>
          <MarketplaceTable />
        </MemoryRouter>
      )

      await waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })
})
