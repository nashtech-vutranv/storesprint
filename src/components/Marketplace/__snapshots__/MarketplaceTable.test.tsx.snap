// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marketplace Table should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll data-table-mh"
        data-scrollselectors=".p-datatable-wrapper"
        pr_id_1=""
      >
        <div
          class="p-datatable-header"
        >
          <div
            class="row"
          >
            <div
              class="d-flex justify-content-start align-items-center"
            >
              <div
                class="btn-mr-1 col-xxl-8 col-xl-8 col-lg-8 col-md-8 col-sm-8 col-8"
              >
                <span
                  class="p-input-icon-right d-block"
                >
                  <input
                    class="p-inputtext p-component"
                    placeholder="common_input_search_placeHolder"
                    value=""
                  />
                </span>
              </div>
              <button
                class="btn-h-95 btn btn-primary"
                type="button"
              >
                common_button_search_label
              </button>
            </div>
          </div>
        </div>
        <div
          class="p-datatable-wrapper"
        >
          <table
            class="p-datatable-table"
            role="table"
          >
            <thead
              class="p-datatable-thead"
            >
              <tr
                role="row"
              >
                <th
                  aria-sort="ascending"
                  class="p-sortable-column p-highlight"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      marketplace_column_header_name
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      marketplace_column_header_currency
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      marketplace_column_header_status
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="p-datatable-tbody"
            >
              <tr
                class="p-datatable-emptymessage"
                role="row"
              >
                <td
                  colspan="3"
                  role="cell"
                >
                  marketplace_table_empty_message
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="p-paginator p-component p-paginator-bottom table-paginator"
        >
          <div
            class="p-paginator-left-content"
          >
            <button
              aria-label="refresh-button"
              class="p-button p-component p-button-text refresh-button p-button-icon-only"
              type="button"
            >
              <span
                class="p-button-icon p-c pi pi-refresh"
              />
              <span
                class="p-button-label p-c"
              >
                Â 
              </span>
            </button>
          </div>
          <button
            class="p-paginator-first p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-2"
            >
              organization_paginator_first
            </span>
          </button>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_previous
            </span>
          </button>
          <span
            class="p-paginator-pages"
          />
          <button
            class="p-paginator-next p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_next
            </span>
          </button>
          <button
            class="p-paginator-last p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-2"
            >
              organization_paginator_last
            </span>
          </button>
          <div
            class="p-dropdown p-component p-inputwrapper dropdown-table-input opacity-md p-disabled p-inputwrapper-filled"
          >
            <div
              class="p-hidden-accessible"
            >
              <input
                aria-haspopup="listbox"
                aria-label="textbox"
                disabled=""
                readonly=""
                type="text"
              />
            </div>
            <div
              class="p-hidden-accessible p-dropdown-hidden-select"
            >
              <select
                aria-hidden="true"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </div>
            <span
              class="p-dropdown-label p-inputtext"
            >
              10
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="dropdown-trigger"
              class="p-dropdown-trigger"
              role="button"
            >
              <span
                class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
              />
            </div>
          </div>
          <span
            style="user-select: none; text-align: center; margin-left: auto;"
          >
            organization_paginator_show_report
          </span>
          <div
            class="p-paginator-right-content"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll data-table-mh"
      data-scrollselectors=".p-datatable-wrapper"
      pr_id_1=""
    >
      <div
        class="p-datatable-header"
      >
        <div
          class="row"
        >
          <div
            class="d-flex justify-content-start align-items-center"
          >
            <div
              class="btn-mr-1 col-xxl-8 col-xl-8 col-lg-8 col-md-8 col-sm-8 col-8"
            >
              <span
                class="p-input-icon-right d-block"
              >
                <input
                  class="p-inputtext p-component"
                  placeholder="common_input_search_placeHolder"
                  value=""
                />
              </span>
            </div>
            <button
              class="btn-h-95 btn btn-primary"
              type="button"
            >
              common_button_search_label
            </button>
          </div>
        </div>
      </div>
      <div
        class="p-datatable-wrapper"
      >
        <table
          class="p-datatable-table"
          role="table"
        >
          <thead
            class="p-datatable-thead"
          >
            <tr
              role="row"
            >
              <th
                aria-sort="ascending"
                class="p-sortable-column p-highlight"
                role="columnheader"
                tabindex="0"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    marketplace_column_header_name
                  </span>
                  <span
                    class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                  />
                </div>
              </th>
              <th
                aria-sort="none"
                class="p-sortable-column"
                role="columnheader"
                tabindex="0"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    marketplace_column_header_currency
                  </span>
                  <span
                    class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                  />
                </div>
              </th>
              <th
                aria-sort="none"
                class="p-sortable-column"
                role="columnheader"
                tabindex="0"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    marketplace_column_header_status
                  </span>
                  <span
                    class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                  />
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="p-datatable-tbody"
          >
            <tr
              class="p-datatable-emptymessage"
              role="row"
            >
              <td
                colspan="3"
                role="cell"
              >
                marketplace_table_empty_message
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="p-paginator p-component p-paginator-bottom table-paginator"
      >
        <div
          class="p-paginator-left-content"
        >
          <button
            aria-label="refresh-button"
            class="p-button p-component p-button-text refresh-button p-button-icon-only"
            type="button"
          >
            <span
              class="p-button-icon p-c pi pi-refresh"
            />
            <span
              class="p-button-label p-c"
            >
              Â 
            </span>
          </button>
        </div>
        <button
          class="p-paginator-first p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-2"
          >
            organization_paginator_first
          </span>
        </button>
        <button
          class="p-paginator-prev p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-3"
          >
            organization_paginator_previous
          </span>
        </button>
        <span
          class="p-paginator-pages"
        />
        <button
          class="p-paginator-next p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-3"
          >
            organization_paginator_next
          </span>
        </button>
        <button
          class="p-paginator-last p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-2"
          >
            organization_paginator_last
          </span>
        </button>
        <div
          class="p-dropdown p-component p-inputwrapper dropdown-table-input opacity-md p-disabled p-inputwrapper-filled"
        >
          <div
            class="p-hidden-accessible"
          >
            <input
              aria-haspopup="listbox"
              aria-label="textbox"
              disabled=""
              readonly=""
              type="text"
            />
          </div>
          <div
            class="p-hidden-accessible p-dropdown-hidden-select"
          >
            <select
              aria-hidden="true"
              tabindex="-1"
            >
              <option
                value=""
              />
              <option
                value="10"
              >
                10
              </option>
            </select>
          </div>
          <span
            class="p-dropdown-label p-inputtext"
          >
            10
          </span>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="dropdown-trigger"
            class="p-dropdown-trigger"
            role="button"
          >
            <span
              class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
            />
          </div>
        </div>
        <span
          style="user-select: none; text-align: center; margin-left: auto;"
        >
          organization_paginator_show_report
        </span>
        <div
          class="p-paginator-right-content"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
